### DISCUSSION TITLE
Implement Text Highlighting Feature for Improved User Interaction

### DISCUSSION THREAD

**Discussion by apsinghdev** [processing/p5.js-web-editor#2661] [General]
### Increasing Access

This Feature aims to highlight all the occurrences of the similar text when a particular text is highlighted in the editor

### Feature request details

currently, editor highlights the selected text only 

![Screenshot 2023-11-05 161316](https://github.com/processing/p5.js-web-editor/assets/109718740/c1bbfb48-95bc-4199-895d-ee9a5529fabc)

idea is to make it like this ( it would be way more helpful in larger files ) -

![pr issue](https://github.com/processing/p5.js-web-editor/assets/109718740/a8f9a264-8be8-463b-880c-4bbdb6bd8097)

Adding a text highlighting feature like this to the p5.js Web Editor can increase access in the following ways:

- Improved Usability: Text highlighting makes the editor more user-friendly, attracting users who may find code editing intimidating.

- Enhanced Learning: Beginners can understand code structure and syntax more easily, making the editor a valuable educational tool.

- Reduced Errors: Highlighting helps users spot and correct mistakes quickly, encouraging them to use the editor confidently.

- Increased Productivity: Users can code more efficiently, completing projects faster, which may lead to more frequent usage.

- Broader Audience: The feature accommodates a diverse range of users, from novices to experienced programmers, expanding the user base.

@raclim , @lindapaiste please share your thoughts on it? I would be happy to add this feature.

**Comment by lindapaiste** [comment]
We should look into the CodeMirror documentation and see if they have an addon for this.  FYI we are currently using CodeMirror v5 and will be updating to v6 at some point in the future.

**Comment by apsinghdev** [comment]
> We should look into the CodeMirror documentation and see if they have an addon for this. FYI we are currently using CodeMirror v5 and will be updating to v6 at some point in the future.

Currently, there is `match-highlighter` as addon in codemirror v5

https://github.com/processing/p5.js-web-editor/blob/42924b95fa6f839d95d883f6de8b9f3adbd03f6d/client/modules/IDE/components/Editor/index.jsx#L30

So I think using this the functionality of highlighting all occurrences of selected text can be built .

for that I would have to configure `highlightSelectionMatches` using `showToken` and `annotateScrollbar`

https://github.com/processing/p5.js-web-editor/blob/42924b95fa6f839d95d883f6de8b9f3adbd03f6d/client/modules/IDE/components/Editor/index.jsx#L118

There may be some other configurations as well to build this; it's just a high-level overview of what could work.

I'm up for working on this issue. please assign me

**Comment by apsinghdev** [comment]
hey @lindapaiste ! sorry to ping you. just wanna ask can I work on it?

**Comment by raclim** [comment]
Thanks for sharing your thoughts on this! I personally don't use text editor features like these too often, so I'm not sure how well this might be used generally. However, I definitely could be the odd one out here and would love to hear more thoughts about this—I'm going to move this to 'Discussions' for now!
